presto-main/src/test/java/com/facebook/presto/sql/relational/TestRowExpressionTranslator.java

```diff
@@ -149,15 +153,19 @@ public void testHiddenFunctionNot()
     public void testBasicOperator()
     {
         String untranslated = "col1 + col2";
-        TypeProvider typeProvider = TypeProvider.copyOf(ImmutableMap.of("col1", BIGINT, "col2", BIGINT));
-        RowExpression specialForm = sqlToRowExpressionTranslator.translate(expression(untranslated), typeProvider);
-
-        TranslatedExpression translatedExpression = translateWith(
-                specialForm,
-                new TestFunctionTranslator(functionAndTypeManager, buildFunctionTranslator(ImmutableSet.of(TestFunctions.class))),
-                emptyMap());
-        assertTrue(translatedExpression.getTranslated().isPresent());
-        assertEquals(translatedExpression.getTranslated().get(), "col1 -|- col2");
+        List<TypeProvider> typeProviders = ImmutableList.of(
+                TypeProvider.copyOf(ImmutableMap.of("col1", BIGINT, "col2", BIGINT)),
+                TypeProvider.copyOf(ImmutableMap.of("col1", INTEGER, "col2", INTEGER)));
```

Comment:
Nice ! This will work better as a [parameterized test](https://testng.org/#_parameters_with_dataproviders). An example from Presto is [here](https://github.com/prestodb/presto/blob/eb94b3f4b83dae444519c618b42b4c853d4380b0/presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestAddNotNullFiltersToJoinNode.java#L277-L284)

Reply:
Thanks, just added